exercises/accumulate/mix.exs

```diff
@@ -0,0 +1,35 @@
+defmodule Accumulate.MixProject do
+  use Mix.Project
+
+  def project do
+    [
+      app: :accumulate,
+      version: "0.1.0",
+      # elixir: "~> 1.8",
+      start_permanent: Mix.env() == :prod,
+      deps: deps(),
+      aliases: aliases()
+    ]
+  end
+
+  def aliases do
+    [
+      compile: ["compile --warnings-as-errors"]
+    ]
+  end
```

Comment:
Is the `--warnings-as-errors` flag intended for student consumption, or merely for the tests? As written, student code will fail on their machines when they run `mix test`. That might actually be okay, but it is a change from what's happening right now. If this is only intended for the tests, the test script could be extended to do a `MIX_ENV=test mix compile --warnings-as-errors` and check the exit code before the `mix test` for each exercise. If we do end up keeping the aliases, the aliases function should be made private with `defp` and moved after into the private function section (I'd say after `defp deps`). I would lean towards not having this alias present in the config to minimize changes to the student experience in this PR.

Reply:
I had put it in initially just to facilitate the individual testing for CI, but one thing I do like about this is that I find students often ignore warnings by the compiler, eg:

```
iex> import Date, only: [utc_today: 0]
Date
iex> utc_today
warning: variable "utc_today" does not exist and is being expanded to "utc_today()", please use parentheses to remove the ambiguity or change the variable name
```

I don't mind having it in, because I think it will help steer students toward better coding practices, but I can see that this is a departure from what is currently being done.

Am willing to remove it if it is a breaking change.

Reply:
The more I think about this one the more I think we should not do it without a wider discussion since a) it does introduce new blockers into the workflow and b) it is not the default config for mix projects.

Reply:
sounds good, will revert